docsrc/workflow.rst

```diff
@@ -0,0 +1,193 @@
+  `Automatic Differentiation Variational Inference (ADVI) algorithm <https://mc-stan.org/docs/reference-manual/vi-algorithms-chapter.html>`_
+  and returns   a :class:`cmdstanpy.CmdStanVB` object which contains
+  an approximation the posterior distribution in the unconstrained variable space.
+
+* The :meth:`cmdstanpy.CmdStanModel.optimize` runs one of
+  `Stan's optimization algorithms <https://mc-stan.org/docs/reference-manual/optimization-algorithms-chapter.html>`_
+  to find a mode of the density specified by the Stan program.  It
+  returns a :class:`cmdstanpy.CmdStanMLE` object.
+
+* The :meth:`cmdstanpy.CmdStanModel.generate` method runs Stan's
+  `generate_quantities method <https://mc-stan.org/docs/cmdstan-guide/standalone-generate-quantities.html>`_
+  which generates additional quantities of interest from a mode. Its take an existing sample as input and
+  uses the parameter estimates in the sample to run the Stan program's ``generated quantities`` block.
+  It returns a :class:`cmdstanpy.CmdStanGQ` object.
+
+  
+Validate, view, export the inference engine outputs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The inference engine results objects 
+``CmdStanMCMC``, ``CmdStanVB``, ``CmdStanMLE`` and ``CmdStanGQ``
+contain the CmdStan method configuration information
+and the location of all output files produced.
+The provide a common set methods for accessing the inference results and metadata,
+as well as method-specific informational properties and methods.objects 
+
+
+By `metadata` we mean the information parsed from the header comments and header row of the
+`Stan CSV files <https://mc-stan.org/docs/cmdstan-guide/stan-csv.html>`_.
+into a :class:`cmdstanpy.InferenceMetadata` object which is exposed via
+the object's `metadata` property.
+
+* The metadata :meth:`cmdstanpy.InferenceMetadata.cmdstan_config`
+  property provides the CmdStan configuration information parsed out
+  of the Stan CSV file header.
+
+* The metadata :meth:`cmdstanpy.InferenceMetadata.method_vars_cols`
+  property returns the names, column indices of the inference engine method variables,
+  e.g.,
+  `the NUTS-HMC sampler output variables <https://mc-stan.org/docs/cmdstan-guide/mcmc-intro.html#mcmc_output_csv>`_
+  are ``lp__``, ..., ``energy__``.
+
+* The metadata :meth:`cmdstanpy.InferenceMetadata.stan_vars_cols`
+  property returns the names, column indices of all Stan model variables.
+  Container variables will span as many columns, one column per element.
+
+* The metadata :meth:`cmdstanpy.InferenceMetadata.stan_vars_dims`
+  property specifies the names, dimensions of the Stan model variables.
+
+The CSV data is assembled into the inference result object.
+CmdStanPy provides accessor methods which return this information
+either as columnar data (i.e., in terms of the CSV file columns),
+or as method and model variables.
+
+The ``draws`` and ``draws_pd`` methods return the sample contents
```

Comment:
These methods should be links, but it's tricky because they exist in multiple places (true for the rest of the methods in this file too). IMO, we should link to the MCMC versions, but very specifically call out that they exist in other functions -- maybe even add to the method docstrings? 

We can add

```rst

        See Also
        --------
        :meth:`cmdstanpy.CmdStanVB.stan_variable`
```

to each docstring, which creates a box like this in the doc:

![image](https://i.imgur.com/hMsNqaS.png)